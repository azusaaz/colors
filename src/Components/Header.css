.Header {
  display: flex;
  justify-content: space-between;
  padding: 0.5rem 1rem;
}

#siteName{
  font-family: 'Cabin', sans-serif;
  text-align: right;
  font-size: 2rem;
  margin-top:0.5rem;
}

.tabs {
  display: flex;

}

.tab {
  margin-top:1rem;
  margin-right:1rem;
}

.tab:hover{
  cursor: pointer;
}

#navRight{
  display: flex;
}

#dia{
  width: 2rem;
  height: 2rem;
}