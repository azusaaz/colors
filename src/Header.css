.Header {
}

.tabs {
  display: flex;

}

.tabs>div {
  margin: 1rem;
}