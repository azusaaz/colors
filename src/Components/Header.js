import React from 'react';
import './Header.css';

const Header = ({currentPage, setCurrentPage}) => {
  
    // Toggle form page and report page
    let onClick=(e)=>{
      setCurrentPage(e.target.getAttribute("data-name"));
    }
    
    // For tab styling change
    var record  = localStorage.getItem('colorRecord');
    let bold= {fontWeight:'bold'};
    let disable = {pointerEvents: 'none', opacity: 0.5};

    return (
      <div className="Header">

        <div className="tabs">

         <div className="tab" id="tab-form" data-name="form" onClick={onClick} 
         style={currentPage==='form' ? bold : null}>Form</div>

         <div className="tab" id="tab-report" data-name="report" onClick={onClick} 
         style={!record ? disable : currentPage === 'report' ? bold : null}>Report</div>

        </div>

        <div id="navRight">
          <img id="dia" src="dia.png" alt=""/>
          <div id="siteName">colors</div>
        </div>
      </div>
    );
  }


export default Header;