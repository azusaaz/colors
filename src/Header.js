import React from 'react';
import './Header.css';

const Header = () => {
  
    return (
      <div className="Header">
        <div className="tabs">
         <div id="tab-form">Form</div>
         <div id="tab-report">Report</div>
        </div>
      </div>
    );
  }


export default Header;