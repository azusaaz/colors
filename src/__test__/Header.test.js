import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Header from '../Components/Header';

configure({adapter: new Adapter()});


it('expect to render Header component',() =>{
  expect(shallow(<Header/>).length).toEqual(1);
})

it('expect to match snapshot',() =>{

  let testCurrentPage= 'form'
  expect(shallow(<Header currentPage={testCurrentPage}/>)).toMatchSnapshot();
})


const clickFn = jest.fn();


it('expect tab click call onclick', () => {
    const component = shallow(<Header setCurrentPage={clickFn} />);
    const getAttribute=()=>{return 'test'}
    component
      .find('#tab-form')
      .simulate('click',{target:{getAttribute}});
    expect(clickFn).toHaveBeenCalled();
});

