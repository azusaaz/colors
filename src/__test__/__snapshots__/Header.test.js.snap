// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    currentPage="form"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="tabs"
        >
          <div
            className="tab"
            data-name="form"
            id="tab-form"
            onClick={[Function]}
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Form
          </div>
          <div
            className="tab"
            data-name="report"
            id="tab-report"
            onClick={[Function]}
            style={
              Object {
                "opacity": 0.5,
                "pointerEvents": "none",
              }
            }
          >
            Report
          </div>
        </div>,
        <div
          id="navRight"
        >
          <img
            alt=""
            id="dia"
            src="dia.png"
          />
          <div
            id="siteName"
          >
            colors
          </div>
        </div>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="tab"
              data-name="form"
              id="tab-form"
              onClick={[Function]}
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Form
            </div>,
            <div
              className="tab"
              data-name="report"
              id="tab-report"
              onClick={[Function]}
              style={
                Object {
                  "opacity": 0.5,
                  "pointerEvents": "none",
                }
              }
            >
              Report
            </div>,
          ],
          "className": "tabs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Form",
              "className": "tab",
              "data-name": "form",
              "id": "tab-form",
              "onClick": [Function],
              "style": Object {
                "fontWeight": "bold",
              },
            },
            "ref": null,
            "rendered": "Form",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Report",
              "className": "tab",
              "data-name": "report",
              "id": "tab-report",
              "onClick": [Function],
              "style": Object {
                "opacity": 0.5,
                "pointerEvents": "none",
              },
            },
            "ref": null,
            "rendered": "Report",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt=""
              id="dia"
              src="dia.png"
            />,
            <div
              id="siteName"
            >
              colors
            </div>,
          ],
          "id": "navRight",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "id": "dia",
              "src": "dia.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "colors",
              "id": "siteName",
            },
            "ref": null,
            "rendered": "colors",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tabs"
          >
            <div
              className="tab"
              data-name="form"
              id="tab-form"
              onClick={[Function]}
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Form
            </div>
            <div
              className="tab"
              data-name="report"
              id="tab-report"
              onClick={[Function]}
              style={
                Object {
                  "opacity": 0.5,
                  "pointerEvents": "none",
                }
              }
            >
              Report
            </div>
          </div>,
          <div
            id="navRight"
          >
            <img
              alt=""
              id="dia"
              src="dia.png"
            />
            <div
              id="siteName"
            >
              colors
            </div>
          </div>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="tab"
                data-name="form"
                id="tab-form"
                onClick={[Function]}
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Form
              </div>,
              <div
                className="tab"
                data-name="report"
                id="tab-report"
                onClick={[Function]}
                style={
                  Object {
                    "opacity": 0.5,
                    "pointerEvents": "none",
                  }
                }
              >
                Report
              </div>,
            ],
            "className": "tabs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Form",
                "className": "tab",
                "data-name": "form",
                "id": "tab-form",
                "onClick": [Function],
                "style": Object {
                  "fontWeight": "bold",
                },
              },
              "ref": null,
              "rendered": "Form",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Report",
                "className": "tab",
                "data-name": "report",
                "id": "tab-report",
                "onClick": [Function],
                "style": Object {
                  "opacity": 0.5,
                  "pointerEvents": "none",
                },
              },
              "ref": null,
              "rendered": "Report",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt=""
                id="dia"
                src="dia.png"
              />,
              <div
                id="siteName"
              >
                colors
              </div>,
            ],
            "id": "navRight",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "id": "dia",
                "src": "dia.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "colors",
                "id": "siteName",
              },
              "ref": null,
              "rendered": "colors",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
